external/avro/src/main/scala/org/apache/spark/sql/avro/SparkAvroDatumReader.scala

```diff
@@ -420,10 +426,27 @@ class SparkAvroDatumReader[T](
         structReader(rowUpdater, in)
         updater.set(ordinal, row)
     } else {
-      (updater, ordinal, in) =>
-        val row = createRow()
-        structReader(new RowUpdater(row), in)
+      val rowPool = ArrayBuffer.empty[InternalRow]
+      val rowUpdaterPool = ArrayBuffer.empty[RowUpdater]
+      val i = reuseIndices.length
+      reuseIndices += 0
+
```

Comment:
These changes did not result in any improvements so were removed later on. Read times are in column F on pg 2 of the benchmark pdf attached to this PR.